### DISCUSSION TITLE
Is it possible to share the Dockerfile?

### DISCUSSION THREAD

**Discussion by noitsnotit** [ultrafunkamsterdam/undetected-chromedriver#1071] [Q&A] [answered]
Hi @ultrafunkamsterdam - Thank you for the work and we appreciate it! 

Just a small request, is it possible to share the Dockerfile you used to build the image [here](https://hub.docker.com/r/ultrafunk/undetected-chromedriver). I was trying to build a docker image to run on the AWS lambda but I have tried many different approaches still cannot get it right. 

I would like to give it a last try and am wondering if you can share the file you used so that we can learn from it. 

Thanks again!

**Comment by emanodame** [comment]
Please comment if anyone has ran this successfully with Lambda or Docker

**Reply by noitsnotit** [reply, marked as answer]
You can use [this](https://hub.docker.com/r/ultrafunk/undetected-chromedriver) image for docker if you are running it on ec2. Lambda is slightly more tricky as its single process,  you need to set some `chromeoptions` to make it work

```
from tempfile import mkdtemp

def handler(event=None, context=None):
    options = uc.ChromeOptions()
    options.binary_location = '/opt/chrome/chrome' # make sure you provide the path where you download the chrome 
    options.add_argument('--headless')
    options.add_argument('--no-sandbox')
    options.add_argument("--disable-gpu")
    options.add_argument("--window-size=1280x1696")
    options.add_argument("--single-process")
    options.add_argument("--disable-dev-shm-usage")
    options.add_argument("--disable-dev-tools")
    options.add_argument("--no-zygote")
    options.add_argument(f"--user-data-dir={mkdtemp()}")
    options.add_argument(f"--data-path={mkdtemp()}")
    options.add_argument(f"--disk-cache-dir={mkdtemp()}")
    options.add_argument("--remote-debugging-port=9222")
```

**Comment by emanodame** [comment]
Do you have an example Dockerfile or repo setup with it ?

**Comment by HoroTW** [comment]
Hey hey,
not the original Dockerfile but I just build my own.
It includes python and deps and also VNC so it should be easy to tailor it to your liking. smile
https://github.com/HoroTW/uc-docker
